### PR TITLE
Use bom for more dependencies including Jackson

### PR DESCRIPTION
**Pull request by chriskilding**
Increment the Jackson version

### CODE REVIEW

**Comment by bitwiseman** [comment]
@chriskilding  We don't require 2.11 to work.  Users may have an older version without ill effect.

**Comment by bitwiseman** [comment]
@chriskilding  Is this change needed for the linked PR?

**Comment by chriskilding** [comment]
Yep it's for the linked PR. Not needed outside of that. It's a test to see how the plugin would handle being fed a newer Jackson version from the BOM; I assume that for conformity you'd want to inherit the Jackson version from the BOM once it's added, rather than override it to 2.9.8.

File: pom.xml [hunk 1]
```diff
@@ -227,7 +227,7 @@
     <java.level>8</java.level>
     <groovy.version>2.4.12</groovy.version>
     <argLine>-Djava.awt.headless=true -Xmx4g -Xms512m -XX:MaxPermSize=512m</argLine>
-    <jackson.version>2.9.8</jackson.version>
+    <jackson.version>2.11.0</jackson.version>
```
Comment (bitwiseman) [inline pom.xml:230]:
```suggestion
    <jackson.version>2.11.2</jackson.version>
```

**Comment by bitwiseman** [comment]
@chriskilding 
Yes, this project would certainly inherit the Jackson version from the bom. 
I'm getting a very strange failure on Windows with this PR. Not sure why yet.  Trying a newer version.  Very likely just brittle tests.

**Comment by chriskilding** [comment]
I've tried upgrading it to BOM v13 and removing the Jackson2-api artifact version locally, which should allow it to be inherited from the BOM. However when I ran `mvn validate` I got disagreements about what it should be from the various Jackson modules that are in play.

There is some discussion about adding more of the Jackson modules to either the BOM or the jackson2-api artifact in https://github.com/jenkinsci/bom/pull/294. This will make our bundled Jackson offering more 'batteries included' or plug and play for downstream authors. I suspect we'd need something like that here to make pipeline-model-definition happy.

**Comment by bitwiseman** [comment]
https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fpipeline-model-definition-plugin/detail/PR-399/22/pipeline this passed.

**Pull request merged**